src/planscape/planning/serializers.py

```diff
@@ -412,4 +445,94 @@ class Meta:
         )
+
+
+class GeoJSONSerializer(serializers.Serializer):
+    geometry = serializers.CharField(required=False)
+    features = serializers.ListField(child=serializers.JSONField(), required=False)
+    type = serializers.ChoiceField(
+        choices=[
+            "Feature",
+            "FeatureCollection",
+            "GeometryCollection",
+            "LineString",
+            "MultiLineString",
+            "MultiPoint",
+            "MultiPolygon",
+            "Point",
+            "Polygon",
+        ]
+    )
+
+    def validate(self, data):
+        geojson_type = data.get("type")
+        if geojson_type == "Feature":
+            self._validate_feature(data)
+        elif geojson_type == "FeatureCollection":
+            self._validate_feature_collection(data)
+        elif "geometry" in data:
+            self._validate_geometry(data["geometry"])
+        return data
+
+    def _validate_feature(self, data):
+        if "geometry" not in data:
+            raise serializers.ValidationError("Feature must have a geometry field.")
+        if "features" in data:
+            raise serializers.ValidationError("Feature cannot have a features field.")
+        self._validate_geometry(data["geometry"])
+
+    def _validate_feature_collection(self, data):
+        if "features" not in data:
+            raise serializers.ValidationError(
+                "FeatureCollection must have a features field."
+            )
+        if "geometry" in data:
+            raise serializers.ValidationError(
+                "FeatureCollection cannot have a geometry field."
+            )
+        for feature in data["features"]:
+            if "geometry" in feature:
+                self._validate_geometry(feature["geometry"])
+
+    def _validate_geometry(self, value):
+        try:
+            GEOSGeometry(json.dumps(value) if isinstance(value, dict) else value)
+        except Exception as e:
+            raise serializers.ValidationError(f"Invalid geometry: {str(e)}")
+
+
+class UploadedScenarioDataSerializer(serializers.Serializer):
+    name = serializers.CharField(max_length=100, required=True)
+    stand_size = serializers.ChoiceField(
+        choices=["SMALL", "MEDIUM", "LARGE"], required=False
+    )
+    planning_area = serializers.IntegerField(min_value=1, required=True)
+    geometry = serializers.JSONField(required=True)
+
+    def validate(self, data):
+        geometry = data.get("geometry")
+        planning_area_id = data.get("planning_area")
+        validated_geometry = self.validate_geometry(geometry)
+
+        if not self._is_inside_planning_area(validated_geometry, planning_area_id):
+            raise serializers.ValidationError(
+                "The uploaded geometry is not within the selected planning area."
+            )
+
+        data["geometry"] = validated_geometry
+        return data
+
+    def validate_geometry(self, value):
+        geojson_serializer = GeoJSONSerializer(data=json.loads(value))
```

Comment:
not sure if it's reasonable to do this, but this seemed like the cleanest option to reuse this serializer and check the incoming data